arangod/Utils/Cursor.h

```diff
@@ -85,6 +92,30 @@ class Cursor {
 
   void setDeleted() { _isDeleted = true; }
 
+  void setLastQueryBatchObject(
+      std::shared_ptr<velocypack::Buffer<uint8_t>> buffer) {
+    _currentBatchResult.second = buffer;
+  }
+
+  Result getLastBatchResult(std::string const& batchId,
+                            VPackBufferUInt8& buffer) {
+    if (_currentBatchResult.first.compare(batchId) != 0) {
```

Comment:
Why compare? :thinking: 
```suggestion
    if (_currentBatchResult.first != batchId) {
```